### ISSUE TITLE
2X different results from identical class with different names (only names)

### ISSUE THREAD

**Issue by alex-jitbit** [dotnet/BenchmarkDotNet#2138]
I have `MyClass` and `MyClass2` which is just copy-pasted same identical code. Again - letter-by-letter same, as in "completely identical" just the names are different.

Occasionally, the results I get (about every run out of 3) are drastically different

| Method |     Mean |    Error |  StdDev |   Gen 0 |   Gen 1 | Allocated |
|------- |---------:|---------:|--------:|--------:|--------:|----------:|
|  Test1 | 463.4 us | 23.23 us | 1.27 us | 49.8047 | 10.7422 |    306 KB |
|  Test2 | 295.2 us | 47.14 us | 2.58 us | 49.8047 | 10.7422 |    306 KB |

The version I use: 0.13.2

**Comment by adamsitnik** [comment]
People open issues like this expecting that they have hit a BenchmarkDotNet bug.

I've investigated multiple similar issues, it took days to do that and the reason behind such differences for "same code" was always either code alignment (#756) or memory alignment (#1587).

#756 was mainly addressed by .NET 6 itself which started aligning the loops: https://www.youtube.com/watch?v=ibHOoIj8U8k

We have not enabled memory randomization (#1587) by default because BDN would perform even more iterations (the results would not be stable so quickly) and it would work only for certain cases (it's not always possible to re-allocate the problematic memory inside global setup method).

My main message here is "it's not a bug, it's alignment. If you don't believe me, use a different tool to verify that".

**Comment by alex-jitbit** [comment]
@adamsitnik  Yes, I understand. But what it means, is that when the code is _ACTUALLY DIFFERENT_ we can not know for sure, was it the code changes, or just the "alignment".
